lib/request-proxy.js

```diff
@@ -36,19 +38,43 @@ var InvalidCheckSumError = TypedError({
     expected: null
 });
 
+var MaxRetriesExceeded = TypedError({
+    type: 'ringpop.request-proxy.max-retries-exceeded',
+    message: 'Max number of retries ({maxRetries}) to fulfill request have been exceeded',
+    maxRetries: null
+});
+
 module.exports = RequestProxy;
 
-function RequestProxy(ringpop) {
-    this.ringpop = ringpop;
+function numOrDefault(num, def) {
+    return typeof num === 'number' ? num : def;
+}
+
+function RequestProxy(opts) {
+    this.ringpop = opts.ringpop;
+    this.retrySchedule = opts.retrySchedule || RETRY_SCHEDULE;
+    this.maxRetries = numOrDefault(opts.maxRetries, this.retrySchedule.length);
+
+    // Simply a convenience over having to index into `retrySchedule` each time
+    this.maxRetryTimeout = this.retrySchedule[this.retrySchedule.length - 1];
+
+    this.retryTimer = null;
```

Comment:
this field cannot be a single value. Multiple `proxyReq()` methods will assign to it at the same time. 

This needs to be an object or array of pending timers. Destroy will have to loop over them.